lib/resource/Directory.js

```diff
@@ -75,7 +77,19 @@ Directory.prototype.getProvider = function getProvider(/* [options,] callback */
     return callback();
   }
 
-  return this.dataStore.getResource(this.provider.href, options, require('./Provider'), callback);
+  function providerFactory(provider) {
```

Comment:
This can be solved like this:

```javascript
function providerFactory(provider) {
  Tenant.super_.apply(this, arguments);
  //...
}
utils.inherits(providerFactory, InstanceResource);
```